fda/extrapolation.py

```diff
@@ -0,0 +1,332 @@
+        """
+
+        pass
+
+    def __call__(self, fdata, eval_points, derivative=0, keepdims=False):
+        """
+        Evaluate points outside the domain range. This function is a wrapper of
+        the method `extrapolate`.
+
+        Args:
+            fdata (:class:´FData´): Object where the evaluation is taken place.
+            eval_points (numpy.ndarray): Numpy array with the evalation points
+                outside the domain range. The shape of the array may be
+                `n_eval_points` x `ndim_image`, where the entry (i,j) will be
+                the coordinate j of the time i, or `nsamples` x `n_eval_points`
+                x `ndim_image`, where the first dimension indicates the sample.
+            derivate (numeric, optional): Order of derivative to be evaluated.
+            keepdims (bool, optional): If the image dimension is equal to 1 and
+                keepdims is True the return matrix has shape
+                nsamples x eval_points x 1 else nsamples x eval_points.
+
+        Returns:
+            (numpy.ndarray): numpy array with the evaluation of the points.
+        """
+
+        return self.extrapolate(fdata, eval_points, derivative=derivative,
+                                keepdims=keepdims)
+
+
+
+class PeriodicExtrapolation(Extrapolator):
+    """Extends the domain range periodically."""
+
+    def extrapolate(self, fdata, eval_points, derivative=0, keepdims=False):
+        """
+        Evaluate points outside the domain range.
+
+        Args:
+            fdata (:class:´FData´): Object where the evaluation is taken place.
+            eval_points (numpy.ndarray): Numpy array with the evalation points
+                outside the domain range. The shape of the array may be
+                `n_eval_points` x `ndim_image`, where the entry (i,j) will be
+                the coordinate j of the time i, or `nsamples` x `n_eval_points`
+                x `ndim_image`, where the first dimension indicates the sample.
+            derivate (numeric, optional): Order of derivative to be evaluated.
+            keepdims (bool, optional): If the image dimension is equal to 1 and
+                keepdims is True the return matrix has shape
+                nsamples x eval_points x 1 else nsamples x eval_points.
+
+        Returns:
+            (numpy.ndarray): numpy array with the evaluation of the points.
+        """
+
+        domain_range = fdata.domain_range
+
+
+        # Extends the domain periodically in each dimension
+        for i in range(fdata.ndim_domain):
+            a, b = domain_range[i]
+            np.subtract(eval_points[..., i], a, eval_points[..., i])
+            np.mod(eval_points[..., i], b - a, eval_points[..., i])
+            np.add(eval_points[..., i], a, eval_points[..., i])
+
+
+
+        # Case evaluation at different times for each sample
+        if fdata.ndim_domain == 1 and eval_points.ndim == 3:
```

Comment:
Why is the one dimensional domain treated independently? I would expect that it could handle as coordinates either numbers or 1-tuples, and that evaluate normalized them to 1-tuples.

Reply:
Changed in e19a1f6. Now with the flag `aligned_evaluation` to indicate the evaluation of all the samples at the same time that treatment is not necessary. 
On the other hand, I'm not sure if this is an adequate name for the parameter.